src/nnet3/nnet-chain-training2.cc

```diff
@@ -284,103 +290,84 @@ void NnetChainTrainer2::ProcessOutputs(bool is_backstitch_step2,
   }
   max_change_stats_.Print(*nnet_);
   return ans;
 }
 
 NnetChainTrainer2::~NnetChainTrainer2() {
-  if (opts_.nnet_config.write_cache != "") {
-    Output ko(opts_.nnet_config.write_cache, opts_.nnet_config.binary_write_cache);
+  if (!opts_.nnet_config.write_cache.empty()) {
+    Output ko(opts_.nnet_config.write_cache,
+              opts_.nnet_config.binary_write_cache);
     compiler_.WriteCache(ko.Stream(), opts_.nnet_config.binary_write_cache);
     KALDI_LOG << "Wrote computation cache to " << opts_.nnet_config.write_cache;
   }
   delete delta_nnet_;
 }
 
-NnetChainModel2::NnetChainModel2(
-    const NnetChainTraining2Options &opts,
-    Nnet *nnet,
-    const std::string &den_fst_dir
-    ):
-    opts_(opts),
-    nnet(nnet),
-    den_fst_dir_(den_fst_dir) {
-}
-
-NnetChainModel2::~NnetChainModel2() {
-}
+NnetChainModel2::NnetChainModel2(const NnetChainTraining2Options& /* unused */,
+                                 Nnet *nnet, const std::string &den_fst_dir)
+    : nnet(nnet), den_fst_dir_(den_fst_dir) {}
 
-NnetChainModel2::LanguageInfo::LanguageInfo(
-    const NnetChainModel2::LanguageInfo &other):
-    name(other.name),
-    den_graph(other.den_graph)
-     { }
+NnetChainModel2::~NnetChainModel2() {}
 
+/* fst::StdVectorFst* NnetChainModel2::GetDenFstForLang( */
+/*        const std::string &language_name) { */
+/*   LanguageInfo *info = GetInfoForLang(language_name); */
+/*   return &(info->den_fst); */
+/* } */
 
-NnetChainModel2::LanguageInfo::LanguageInfo(
-    const std::string &name,
-    const fst::StdVectorFst &den_fst, 
-    int32 num_pdfs):
-    name(name),
-    den_graph(den_fst, num_pdfs){
+const chain::DenominatorGraph *NnetChainModel2::GetDenGraphForLang(
+    const std::string &lang) {
+  const LanguageInfo *info = GetInfoForLang(lang);
+  KALDI_ASSERT(info != nullptr);
+  return &(info->den_graph);
 }
 
-void NnetChainModel2::GetPathname(const std::string &dir,
-                                   const std::string &name,
-                                   const std::string &suffix,
-                                   std::string *pathname) {
-  std::ostringstream str;
-  str << dir << '/' << name << '.' << suffix;
-  *pathname = str.str();
+namespace {
+
+// Get a pathname in the form '<dir>/<name>.<suffix>'.
+std::string GetPathname(const std::string &dir,
+                        const std::string &name,
+                        const std::string &suffix) {
+  return dir + '/' + name + '.' + suffix;
 }
 
-void NnetChainModel2::GetPathname(const std::string &dir,
-                                   const std::string &name,
-                                   int32 job_id,
-                                   const std::string &suffix,
-                                   std::string *pathname) {
-  std::ostringstream str;
-  str << dir << '/' << name << '.' << job_id << '.' << suffix;
-  *pathname = str.str();
+// TODO(kkm): Currently unused. Delete or use if intended.
+#if 0
```

Comment:
delete/cleanup?
Otherwise looks OK.

Reply:
It's unused and is fairly trivial 2-line long helper, so I better kill it. I'm no fan of dead code, and many times so when it's trivial, 2-line long, and a helper whose help is unasked for. The rest I'm merging then.